### PR TITLE
Test/years of experience

### LINKED ISSUE #1009: Add "Years of experience" field to onboarding form
Note : This issue will only be assigned to beginners.
If you have previously contributed to open source, or have experience, please filter out the issues tagged with "good first issue" and "beginner friendly".

## Context

On slide three of the additional-details (onboarding form) the user is presented with a dropdown, from which they select whether they are a "Working professional", "Current student" or "None of the above".

If they select "Working professional" they are presented with two further questions. 
"Where are you working?" and "What is your job title?".
Additionally to this we would like to ask for "Years of experience".
A dropdown would work, but you could alternatively add a text input.
Either way please include Zod schema for the validation of a reasonable number.
Please use the catalyst components and remain consistent with the form styling.
A new field will be required on the user table. 
 
## Current Behaviour

Current Slide three

## Additional info

Note: The catalyst components can be found at :

components> ui-components

The onboarding form is found at

app>(app)>alpha>additional-details

Any questions feel free to ask.

Connected issues
#1003
#1004 
#1005 
#1006 
#1008 
#1009
#1011

### PR DESCRIPTION
**Pull request by leslyclopez11**
# ✨ Codu Pull Request 💻

Fixes #1009

## Pull Request details

- I added code to _client.tsx modifying slideThree (including the dropdown)
- Defined yearsOfExperience in additionalUserDetails.tsx
- Defined yearsOfExperience in page.tsx
- Defined yearsOfExperience in schema.ts
- I added validation based on coderabbitai suggestions
- Attempted to create the Drizzle migration
- I cherry-picked from the develop branch the commits, but I'm confused if I'm doing this right.

## Any Breaking changes

- 'None'

## Associated Screenshots

- 'None'

### CODE REVIEW

**Comment by John-Paul-Larkin** [comment]
Hey @leslyclopez11 
Great to see you back! I am sure you learned loads about Git 🤣 
I see you have two PRs open now. 
Test/years of experience #1121 and Feature/years of experience #1120

Which one of these can be closed?

**Comment by leslyclopez11** [comment]
@John-Paul-Larkin oh, I fear I did learn a lot of git. Hopefully, I did the PR correct now but let me know if that's not the case. I feel more confident with this one: Test/years of experience https://github.com/codu-code/codu/pull/1121

**Comment by John-Paul-Larkin** [comment]
> @John-Paul-Larkin oh, I fear I did learn a lot of git. Hopefully, I did the PR correct now but let me know if that's not the case. I feel more confident with this one: Test/years of experience #1121

Great, so I will close the other PR and review this one. 
PS. Git is complicated and very powerful, but it will all start to make sense to you soon!

**Comment by John-Paul-Larkin** [comment]
Hey @leslyclopez11 
This is great work. Thanks very much.
You are nearly there with this PR, but there are a few small things we should fix first.

The value '0-1' is defaulted to for every record. 
The database is not actually being update with the entered value. 
You can search for the function "slideThreeSubmitAction", which is in the _actions file in the additional-details directory.
This is a Next.js server action and is invoked to update the database. 
You can follow the pattern to update the database with the yearsOfExperience value.

On the same subject, we actually dont want a default value for this. The field should be null until the user inputs something in the form.

If you run "npx drizzle-kit studio" it will connect to your database and spin up proxy server for Drizzle Studio which you can use for viewing the database.

I hope this all makes sense?

**Comment by NiallJoeMaher** [comment]
Is this still being worked on? Just so I know if I should keep it open or close?

**Comment by leslyclopez11** [comment]
@NiallJoeMaher I'm still working on this!

File: drizzle/0010_gorgeous_living_tribunal.sql
Comment (NiallJoeMaher) [inline drizzle/0010_gorgeous_living_tribunal.sql:1]: There should be only one migration file for the updates. Let me know if you still want to take this on or else we can take it over.

**Review by NiallJoeMaher — CHANGES_REQUESTED** [review]

**Pull request closed without merging**